pip/commands/search.py

```diff
@@ -131,3 +134,39 @@ def print_results(hits, name_column_width=None, terminal_width=None):
 
 def highest_version(versions):
     return max(versions, key=parse_version)
+
+def sort_hits(query, hits):
+    if len(query) != 1:
+#   Now only handle the easiest condition,
+#   which is possibly the most frequent condition
+        return hits
+    query = query[0]
+    p = re.compile(query)
+    p_ignorecase = re.compile(query, re.IGNORECASE)
+    for index, item in enumerate(hits):
+        hits[index]['match_score'] = 0
```

Comment:
for any use of `hits[index]` you can just use `item`

the enumerate is also unnecessary then